lib/routes/shared/security_pin/change_pin_code.dart

```diff
@@ -49,24 +49,26 @@ class _ChangePinCodeState extends State<ChangePinCode> {
       _setPinCodeInput(_enteredPinCode + numberText);
     }
     if (_enteredPinCode.length == PIN_CODE_LENGTH) {
-      if (_tmpPinCode.isEmpty) {
-        setState(() {
-          _tmpPinCode = _enteredPinCode;
-          _label = "Re-enter your new PIN";
-          _enteredPinCode = "";
-        });
-      } else {
-        if (_enteredPinCode == _tmpPinCode) {
-          Navigator.pop(context, _enteredPinCode);
-        } else {
+      Future.delayed(Duration(milliseconds: 200), () {
```

Comment:
@erdemyerebasmaz instead of duplicating this code over all pin code variations what do you think about changing the `onNumButtonPressed` to be called after the animation delay?
This way we also don't need to guess the duration here.

Reply:
If we add it to ```onNumButtonPressed```, there's nothing to be animated because _enteredPinCode has not changed yet.
We only need to delay when the _enteredPinCode.length == PIN_CODE_LENGTH condition is met because that's where we reset the pin input/navigate elsewhere.

Reply:
I see...
In that case I think the PinCodeWidget should handle all the "single letter" events as it already knows the required length and render all the circles.
How about removing the `onNumButtonPressed` event and only expose `onPinEntered` event after the animation?
This will reduce some duplicate code from all the three pin code widgets and will require to convert the PinCodeWidget to be stateful.

Reply:
Fixed 👍